services/app/assets/js/widgets/components/UserInfo.jsx

```diff
@@ -1,43 +1,46 @@
 import React from 'react';
-import {OverlayTrigger, Popover} from 'react-bootstrap';
+import { OverlayTrigger, Popover } from 'react-bootstrap';
 import Loading from './Loading';
 import UserName from './UserName';
-import {loadUserStats} from '../middlewares/Users';
 
-const UserInfo = ({user, usersStats}) => {
-  let statsContent = <Loading small={true}/>;
-
-  const stats = usersStats[user.id];
+const renderStatsPopoverContent = (stats) => {
   if (stats) {
-    statsContent = (
+    return (
       <div>
-        Won: <b className="text-success">{stats.won}</b>
-        <br/>
-        Lost : <b className="text-danger">{stats.lost}</b>
-        <br/>
-        Gave up : <b className="text-warning">{stats.gave_up}</b>
+        Won:
+        <b className="text-success">{stats.won}</b>
+        <br />
+        Lost:
+        <b className="text-danger">{stats.lost}</b>
+        <br />
+        Gave up:
+        <b className="text-warning">{stats.gave_up}</b>
       </div>
     );
   }
 
+  return <Loading small />;
+};
+
+const UserInfo = ({ user, usersStats, loadUserStats }) => {
   const statsPopover = (
     <Popover title="Stats">
-      {statsContent}
+      {renderStatsPopoverContent(usersStats[user.id])}
     </Popover>
   );
 
-  const onEnter = () => typeof stats === 'undefined' ? loadUserStats(user) : null;
+  const onEnter = () => (typeof stats === 'undefined' ? loadUserStats(user) : null);
```

Comment:
Does `stats === undefined` work instead of `typeof stats === 'undiefined'` ?

Reply:
Wow, never compare with undefined. Just try ```stats ? null : loadUserStats(user)```